verifying throws NPE when arg to mocked fn in spec is an anon fn

dchelimsky opened the pull request:
This PR addresses #4, supporting anonymous functions as function arguments in `verifying` expressions.

Additionally:
* added namespaces to internal data structures for clarification
* removed verification-specific keys when only providing

fernando-nubank commented:
> So far, this just has a failing test to expose the bug.

Best way to start fixing a bug ;)

dchelimsky commented:
/cc @pmatiello - just FYI

fernando-nubank reviewed (approved):
:shipit:

dchelimsky committed:
support anon fns as args when verifying

Additional refactoring:
- added namespaces to internal data structures for clarification
- removed verification-specific keys when only providing
